x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilegeResolver.java

```diff
@@ -155,6 +155,9 @@
     public static final NamedClusterPrivilege CANCEL_TASK = new ActionClusterPrivilege("cancel_task",
         Set.of(CancelTasksAction.NAME + "*"));
 
+    public static final NamedClusterPrivilege ENROLL_NODE = new ActionClusterPrivilege("enroll_node",
+        Set.of("cluster:admin/xpack/security/enrollment/enroll/node"));
```

Comment:
I remember we discussed this being excluded from `manage_security`. I would also suggest this also be excluded from `all`.
Human users do not need the `enroll_node` privilege, as it should only be assigned to trusted setup scripts. It should also only be assigned to credentials by another setup script. In other words, there is no reason, apart from convention, for human users with `manage_security` or `all` privileges to get access to the enroll node API in v8. If anything, the `enroll_node` should "contains" the `all` and `manage_security` privileges.

IMO this is a special API with a special privilege, and we should treat it as an exception to the conventions, but I'm going to quickly ask the hive mind as well.

Reply:
I realize that the `all` suggestion might be contentions.
I'm OK with excluding it from `manage_security` only, for the moment, and possibly rectifying `all` in a follow-up.

Reply:
We discussed on another channel. We're not going to exclude anything because it is not improving security because users with `all` and `manage_security`, even if they don't have the new privilege themselves, can add new users that can indeed have the new privilege.
Later on, we're going to think of ways to prevent this.